.github/workflows/tests_linux.yml

```diff
@@ -486,4 +486,61 @@ jobs:
         uses: codecov/codecov-action@v3
         with:
           fail_ci_if_error: true
-          token: ${{ secrets.CODECOV_TOKEN }}
\ No newline at end of file
+          token: ${{ secrets.CODECOV_TOKEN }}
+
+  cpptestsWithMultipleBackends:
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        exec_model: [\"SERIAL\"]
+        kokkos_version: ${{ fromJson(needs.build_and_cache_Kokkos.outputs.kokkos_version) }}
+
+    name: C++ tests
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.9'
+
+      - name: Checkout PennyLane-Lightning
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Restoring cached dependencies
+        id: kokkos-cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace}}/Kokkos_install/${{ matrix.exec_model }}
+          key: ${{ matrix.os }}-kokkos${{ matrix.kokkos_version }}-${{ matrix.exec_model }}
+
+      - name: Copy cached libraries
+        run: |
+          mkdir Kokkos/
+          cp -rf ${{ github.workspace}}/Kokkos_install/${{ matrix.exec_model }}/* Kokkos/
+
+      - name: Install dependencies
+        run: sudo apt-get update && sudo apt-get -y -q install cmake gcc-$GCC_VERSION  g++-$GCC_VERSION  ninja-build gcovr lcov
+
+      - name: Build and run unit tests
+        run: |
+            cmake . -BBuild -G Ninja \
+              -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+              -DBUILD_TESTS=ON \
+              -DCMAKE_PREFIX_PATH=${{ github.workspace }}/Kokkos \
+              -DENABLE_PYTHON=OFF \
+              -DPL_BACKEND="lightning_qubit;lightning_kokkos" \
+              -DCMAKE_CXX_COMPILER=$(which g++-$GCC_VERSION)
+            cmake --build ./Build
+            cd ./Build
+            mkdir -p ./tests/results
+            for file in *runner ; do ./$file --order lex --reporter junit --out ./tests/results/report_$file.xml; done;
```

Comment:
Yes, you're right! but device agnostic tests cannot be built with multiple-devices and it may not be required for the scope of this work. What do you think?

Reply:
The best course of action would be to test the device-agnostic part with both.
The second best would be to keep it like this but add at least a comment in the CI clarifying that we are not testing both in the device-agnostic tests, just LightningQubit. I can live with the second-best in this case.
I will make a suggestion for such a comment.

Reply:
Thanks @AmintorDusko!